### PR TITLE
feat (workspace widget): Added new functionality to widgets modal

### PR DESCRIPTION
**Pull request by pavlovicnemanja92**
SD-2415

### CODE REVIEW

File: client/app/scripts/superdesk-dashboard/module.js [hunk 1]
```diff
@@ -36,13 +36,18 @@ define([
...
         };
 
         this.selectWidget = function(widget) {
-            this.selectedWidget = widget || null;
+            if (!this.isSelected(widget)) {
+                this.selectedWidget = widget || null;
+            }
+        };
+
+        this.isSelected = function (widget) {
```
Comment (petrjasek) [inline client/app/scripts/superdesk-dashboard/module.js]: pls add docstring

File: client/app/scripts/superdesk-dashboard/module.js [hunk 2]
```diff
@@ -36,13 +36,18 @@ define([
 
         this.addWidget = function(widget) {
             this.widgets.push(widget);
-            this.availableWidgets = getAvailableWidgets(this.widgets);
             this.selectWidget();
             this.save();
         };
 
         this.selectWidget = function(widget) {
```
Comment (petrjasek) [inline client/app/scripts/superdesk-dashboard/module.js]: +docstring

**Comment by ioanpocol** [comment]
:+1:

**Comment by petrjasek** [comment]
:+1:

**Comment by plamut** [comment]
Tests are missing for the changes in `DashboardController`. The rest looks good.

**Comment by pavlovicnemanja92** [comment]
There are no tests for this controller (for widgets at all) ... I think it will take some time for me to write tests for all the cases. Maybe is good to create task just for writing tests for widgets.

**Comment by pavlovicnemanja92** [comment]
According to talk with Ioan, I created a task for writing all test for widgets: https://dev.sourcefabric.org/browse/SD-2828

**Pull request merged**